### PR TITLE
Ben Fulton

### PR DESCRIPTION
**Pull request by b-enji-cmd**

### CODE REVIEW

File: lib/network.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  def add_show(show)
+    @shows << show
+  end
+
+  def main_characters
+    @shows.flat_map do |show|
+      show.characters.select do |character|
+        character.name == character.name.upcase
```
Comment (b-enji-cmd) [inline lib/network.rb:17]: Rats, forgot to paste back in the salary requirement...

File: lib/character.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Character
+  attr_reader :name,
+              :actor,
+              :salary
+
+  def initialize(details)
+    @name = details[:name]
+    @actor = details[:actor]
+    @salary = details[:salary]
```
Comment (mikedao) [inline lib/character.rb:9]: good

File: lib/network.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      end
+    end
+  end
+
+  def actors_by_show
+    show_breakdown = Hash.new { |h, k| h[k] = [] }
```
Comment (mikedao) [inline lib/network.rb:23]: This is unneccessary

File: lib/network.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  def shows_by_actor
+    actor_breakdown = Hash.new { |h, k| h[k] = [] }
+
+    @shows.map do |show|
+      test = show
+      show.characters.map do |character|
+        actor_breakdown[character.actor] << test if test.actors.include?(character.actor)
+      end
+    end
+    actor_breakdown
```
Comment (mikedao) [inline lib/network.rb:39]: Method does a bit much, would push logic down to a lower method

File: lib/network.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  def prolific_actors
+    actors = shows_by_actor
```
Comment (mikedao) [inline lib/network.rb:43]: unnecessary variable assignment

File: lib/show.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+              :characters
+
+  def initialize(name, creator, characters = [])
+    @name = name
+    @creator = creator
+    @characters = characters
+  end
+
+  def total_salary
+    @characters.sum(&:salary)
```
Comment (mikedao) [inline lib/show.rb:13]: Use the multiline syntax

**Review by mikedao — COMMENTED** [review]
Overall good job. Stick to using multi-line syntax.

Score: 3.5

**Pull request closed without merging**